DMX.Core.Api.Tests.Unit/Services/Foundations/Labs/LabServiceTests.Validations.Remove.cs

```diff
@@ -0,0 +1,103 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// ---------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using DMX.Core.Api.Models.Foundations.Labs;
+using DMX.Core.Api.Models.Foundations.Labs.Exceptions;
+using FluentAssertions;
+using Moq;
+using Xunit;
+
+namespace DMX.Core.Api.Tests.Unit.Services.Foundations.Labs
+{
+    public partial class LabServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowValidationExceptionOnRemoveIfIdIsInvalidAndLogItAsync()
+        {
+            // given
+            var invalidId = Guid.Empty;
+            var invalidLabException = new InvalidLabException();
+
+            invalidLabException.AddData(
+                key: nameof(Lab.Id),
+                values: "Id is required");
+
+            var expectedLabValidationException =
+                new LabValidationException(invalidLabException);
+
+            // when
+            ValueTask<Lab> removeLabTask = this.labService.RemoveLabByIdAsync(invalidId);
+
+            LabValidationException actualLabValidationException =
+                await Assert.ThrowsAsync<LabValidationException>(
+                    removeLabTask.AsTask);
+
+            // then
+            actualLabValidationException.Should().BeEquivalentTo(
+                expectedLabValidationException);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                    expectedLabValidationException))),
+                        Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectLabByIdAsync(It.IsAny<Guid>()),
+                    Times.Never);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.DeleteLabAsync(It.IsAny<Lab>()),
+                    Times.Never);
+
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowValidationExceptionOnRemoveIfLabIsNullAndLogItAsync()
+        {
+            // given
+            Lab nullLab = null;
+            Lab returnedLab = nullLab;
```

Comment:
Remvoe both of these variables and replace them with just one variable that says `noLab`

```csharp
Lab noLab = null;
```

So when we are doing a `ReturnsAsync` it would look nice: "returns no lab"